### ISSUE TITLE
Go to file only shows some files (and I can't switch folders)

### ISSUE THREAD

**Issue by bellebethcooper** [careteditor/caret#409]
In 3.0.1 I only see some of the files in the current folder when using CMD + T to go to file, though they all seem to be shown when using the sidebar.

I also can't figure out how to switch folders when I'm in the file browser view. I used to see the previous folder (higher up in the hierarchy) in the list of files, and choosing it would move me into that folder and show its files.

**Comment by slamprecht** [comment]
Exactly same here. Just some files, not way to navigate through folders. Just stucked in the actual one. Even if u save a file manually in a different location, I am stucked there.

**Comment by astoilkov** [comment]
We actually changed the way file navigator `Cmd/Ctrl+T` works. Now the file navigator searches in all subdirectories but doesn't allow you to go up the file system hierarchy to find a file.

Previously the file navigator would search only in files and folder names that are in the current folder. This way you would navigate through folders to find your file. 

We are considering two options for bringing the previous functionality back:

- make another popup that works exactly the same way as the previous solution
- integrate folder navigation in the current file navigator while keeping the search in all subdirectories

If anyone have ideas please let us know.

**Comment by slamprecht** [comment]
Hi, there,

first of all, I think there are 2 problems.

1. Despite your change of functionality. As I showed you in my Twitter Post, at moment cmd-t just shows a few files in a directory with lots more. Without any possibility to navigate in the current folder. That sounds like a bug for me. Not a new feature. :-)
2. I don't understand the change of functionality you described. I am an experienced user and it-writer for over 25 years in industry, and in the moment i can't imagine any case, where the behaviour you describe would be useful. For me the big advantage of caret and the behaviour before was, that without touching a mouse or trackpad, I could save a file exactly in file system where I want. So get up and down in file hierarchy as I wish. Why should any user just want to stay in one folder? That sounds not useful to me. 

So I vote for bringing back the "old" functionality, that not only me but also other bloggers and writers welcomed. To develop another popup would make caret more complex. You would have sidebar, the cmd-t panel and the "new" cmd-t Panel with Navigation. The asset for caret so far was the focus on writing and assistance, and not to be complex. 

Just my 2 Cents

Regards
Stephan

**Comment by bellebethcooper** [comment]
I agree with Stephan that I can't come up with a reason this new approach would be useful, but I have no doubt you had a good reason for designing it that way. I'm happy with any approach that brings back folder navigation, but for now I'm planning on switching to a different app for my daily writing needs due to the changes in 3.0

**Comment by slamprecht** [comment]
I totally agree with you. As a professional writer that so far produced at least 4 articles a day for different customers and projects (= folders), the new enhancement is a big step backwards for me and my productivity. 

Regards
Stephan

**Comment by borekb** [comment]
I actually like the new approach but agree that it would be useful to still be able to navigate, like you suggest, @astoilkov:

> integrate folder navigation in the current file navigator while keeping the search in all subdirectories

**Comment by slamprecht** [comment]
Any News about it? Can you assign it at least as a bug? Cmt-t just shows here on my mac a selection of files. I can see some JPGs and the arrow keys let me just go up and down in this selection. The folder itself contains lot more files. Despite the "functionality" not having the option to navigate. At least the command should be able to show all files in current directory. What it doesn't...

**Comment by erusev** [comment]
We're working on it, we'll release an update early next week.

**Comment by slamprecht** [comment]
I can see now all files in the folder that I am locked in, after cmd+t. So the bug can be closed, but I still can't see in which sense the loss of navigation could be an enhancement. 

Regards
Stephan

**Comment by erusev** [comment]
> In 3.0.1 I only see some of the files in the current folder when using CMD + T to go to file, though they all seem to be shown when using the sidebar.

This is fixed now.

> I also can't figure out how to switch folders when I'm in the file browser view. I used to see the previous folder (higher up in the hierarchy) in the list of files, and choosing it would move me into that folder and show its files.

We're transitioning to a different approach to file management. I'll try to explain the thinking behind it and hopefully, I'll convince you that it makes sense.

Before 3.0, the file browser opened on cmd+t and showed on top of the open document. We found that this is not optimal as a lot of people wanted to be able to see the files in their project (folder) alongside their document. To resolve this, we decided to move the file browser to a sidebar and give it the more common shortcut, cmd+1.

We also found that a lot of people work on project that contain folders (subfolders) and that it's hard for them to browse subfolders and jump from a file in one subfolder to a file that's in another one. To resolve this, we borrowed a concept from code editors that lets people open a folder (File › Open Folder) and have all subfolders of that folder available to them in the file browser.

Since 3.1 we also have "Go to Folder" cmd+shift+e that helps people quickly move from one project folder to another.

The "popup" that now opens on cmd+t is not a file browser. It's a file search. The idea here is that there are occasions where you know the name of the file you want to open and instead of browsing to find the file, you'd rather just (fuzzy) search for it.

This sums it up.

Do you @bellebethcooper @slamprecht  @borekb think this new approach to file management makes sense and do you think you can adapt your workflows to it?

Thanks a lot!

**Comment by slamprecht** [comment]
Sorry to say, but I can't agree with you. In my version the commands you describe is: "go to recent folder", "go to recent file". All what u say makes sense for long copy writer or even coders, that work in "projects" with a lot of files. I don't. I write 3-5 articles a day. But in my file hierarchy. the project approach was exactly what makes me leave ulysses and go to caret. And now you do the same. If I could at least navigate in "recent folder". But it shows me the ... recent folders. not my folder structure. But..what when I now have to write a new text for another client, that is saved to steps higher in hierarchy? In others words: Caret now tries to make me adopt a workflow that u developed. Therefore it was a tool and let me do my workflow. So..I really was so enthusiastic about the programme and reviewed it very enthusiastic (www.der-lifehacker.de). But it seems to be to leave it already. Cause I want to see my files in total, not "projects" that I have to organize.  

at last: I know I am saying it again. You have now 3 modes to let me see files (or find files): recent folders, recent files, search box. There is poor documentation and if u rethink it: You think that many people, or better users, will understand the concept, that there are 3 ways to achieve the same result? But all 3 lack the natural process that u learn from begining under mac and windows:  in finder and explorer - to move up and down your folders? After 25 years in industry - thats not really usability. A file management without navigation is no file management. Its a condensed view of MY structure. But it lacks, what its intended - the management. 

Regards
Stephan

**Comment by erusev** [comment]
@slamprecht 

Thanks for the quick response!

> Caret now tries to make me adopt a workflow that u developed. 

Our intention has never been to create a specific workflow for people to follow. I understand that the transition from 2.0 to 3.0 can be a bit confusing but the ideas in 3.0 are very simple.

> You have now 3 modes to let me see files (or find files): recent folders, recent files, search box.

When you put it this way it sounds complicated but it really isn't. It's just a file browser, like the one in Finder, a search tool, like the one in Finder and recent folders, like Go › Recent Folders in Finder. The main difference from Finder is that our file management can be constrained to a scope or a project folder so that browsing and searching is faster and more convenient. This mean that you can have a full-featured file manager that doesn't have to think about folders that have nothing to do with your documents (like the Music or Movie folders).

We're releasing version 3.2 next week that should make things even simpler and more consistent.

I appreciate your feedback and I'd love to know if you have more thoughts.

If you prefer, we can talk over email - we are at contact [at] caret.io

Thanks!

**Comment by bellebethcooper** [comment]
I'm finding it a bit confusing to understand all this, but I think I kind of get it now...

It sounds like cmd+t is now a bit like cmd+shift+o in many other apps (i.e. quick open, with fuzzy search). I LOVE the quick open feature in other apps, and would love to use it in Caret, though I don't want it to be constrained to the folder I'm in, as I'm often switching between files in different folders (e.g. files in a folder for my day job and files in other folders related to personal projects). But it seems like right now it's only a quick open for files in the current folder.

I really like the cmd+1 and cmd+2 sidebar shortcuts, but I'd like to be able to go up and down the folder hierarchy in the sidebar, the way I used to be able to with cmd+t. I'm still confused about how the new features support opening files that are in different folders, unless maybe I'm meant to just use my Mac's Finder in those cases?

**Comment by erusev** [comment]
@slamprecht 

What if we make File › Open (cmd+o), default to the current folder? From there you'd be able to hit cmd+up to go to the parent folder, type to jump to an item of your choice and basically do most of the things that the old cmd+t had to offer.

**Comment by slamprecht** [comment]
Hi, 
 I really appreciate the time you invest to me thoughts. At moment for me it's hard to see, what is the intention of the sidebar and what might be a bug.

1. When I hit cmd+1 the sidebar does not open reliable after 1st shortcut. It occurs quite often that I have to repeat input. Also to closed it again.
2. I open a document folder in it with cmd+shift-o. Okay now Isee the folder in it. If I choose a folder on very high position of my hierarchy I see now all folders in it and can choose one that already exist. When I now use cmd-n I can create a new file. But saving it presents me the old finder dialogue. I was so happy not to see it anymore, cause in search I could create files and folder just with the magic of cmd+enter. But anyway. So I can create a new file though all in caret would be my wish.
3. Not I detect the context menu to the items in sidebar. "Create new folder". So that would be needed for a true filemanagement. But when I use this command from context menu (right click), creating a new folder does...nothing. The view on sidebar seems to refresh and my cursor is two lines ahead in my actual open document. Bug? Feature?

I am not sure if Your idea would really help. I can live with a documented command "open folder", cause I decide then what to open. What does "current folder" mean? I mean, I do know what u intent, but does the usual user know what current means? I think that would be confusing, cause the function would open things, he might now intent to open. 

So if u decide to use an command "open folder"  as its now used in the sidebar. You should implement the functionality to let me create new folders in it on any position under the 1st one I opened.

Working with sidebar:

cmd-shift-o

--->
choosing ProjectA
--->
arrows let me see:

ProjectA
- Folder 1
- Folder 2
Project B
and so on

command-arrow-up: I could switch to the parents and grandparents of my 1st choosen ProjectA. 

But without the possibility to create a new folder in the hierarchy its still a view and I need the finder. But when I have to open it anyway, why should I use caret for the project management?

Regards
Stephan

**Comment by erusev** [comment]
@slamprecht 

Thanks for the quick response!

> When I hit cmd+1 the sidebar does not open reliable after 1st shortcut. It occurs quite often that I have to repeat input. Also to close it again.

When the sidebar is open and the focus is on the editor, cmd+1 focuses on the sidebar. Once the sidebar is on focus, hitting cmd+1 again will close it. This means that you can use the same shortcut that opens the sidebar to focus on it. This is a common behavior in code editors.

> When I now use cmd-n I can create a new file. But saving it presents me the old finder dialogue. I was so happy not to see it anymore, cause in search I could create files and folder just with the magic of cmd+enter.

What if when the focus is in the file browser sidebar, you could use cmd+n to create a file or a folder in the current directory?

> But when I use this command from context menu (right click), creating a new folder does...nothing. The view on sidebar seems to refresh and my cursor is two lines ahead in my actual open document. Bug? Feature?

This sounds like a bug - we'll investigate.

Next week we are releasing 3.2 that will make it possible to type to jump in the file browser sidebar and some other improvements that should make the interactions with the file system better.

**Comment by slamprecht** [comment]
Hi there, 

I can't reproduce what you tell me.

If the focus is in editor cmd+1 opens the sidebar. and there the focus remains at moment. anyway how often I hit cmd+1 the sidebar remains open until I click(!) back in editor.

Cmd-N let me produce obivously a new file. But in the folder that I opened before with cmd-shift-o. NOT in the folder that is selected in the sidebar. Maybe thats the misunderstanding? I would like to create a folder via sidebar in the position of the hierarchy I am in. 

I can't figure out how to create a folder. cmd+n creates a new document. How do I transform it into a folder?

And I just tried it again. The context menu (create file, create folder) in sidebare does exactly nothing. 

Regards
Stephan

**Comment by borekb** [comment]
I generally like the new behavior but am missing a way to go up one folder easily (I'm on 3.2 now). What you suggest in https://github.com/careteditor/caret/issues/409#issuecomment-313067254 is not a great UX on Windows, IMO (I prefer the neat Ctrl+T over the system File Open dialog).

I get that the sidebar is sort of a project and changing the base folder too easily doesn't fit the general concept, however, I think that Ctrl+T could incorporate this somehow. I don't have a suggestion I'd love but for example, as a bit of a workaround, if I manually typed `../`, it would scope the search to a parent directory (same could work for subdirectories, e.g., `./subdirectory/...`). Just an idea, "go up" would be generally very useful to me.
